File: Typescript/src/isBalanced.ts
export function isBalanced(num: string): boolean {
	let even = 0;
	let odd = 0;
	for (let i = 0; i < num.length; i += 2) {
		even += parseInt(num.charAt(i));
	}
	for (let i = 1; i < num.length; i += 2) {
		odd += parseInt(num.charAt(i));
	}
	return even === odd;
};

File: Dart/lib/isBalanced.dart
class Solution {
  bool isBalanced(String num) {
    int left = 0, right = 0;
    for (int i = 0; i < num.length; i += 2) {
      left += int.parse(num[i]);
    }
    for (int i = 1; i < num.length; i += 2) {
      right += int.parse(num[i]);
    }
    return left == right;
  }
}
